node/lib/statsd.js

```diff
@@ -36,53 +39,88 @@ function TChannelStatsd(channel, statsd) {
 }
 
 function clean(str, field) {
+    var copy;
+
     if (!str) {
-        return 'no-' + field;
-    } else {
-        // TODO generalize
-        return str
-            .replace(/:/g, '-')
-            .replace(/\//g, '-')
-            .replace(/\./g, '-')
-            .replace(/{|}/g, '-');
+        return field;
     }
+
+    copy = LENGHT_ARRAYS[str.length];
+    if (!copy) {
+        copy = LENGHT_ARRAYS[str.length] = [];
+    }
+
+    for (var i = 0; i < str.length; i++) {
+        var char = str[i];
+
+        if (char === ':' ||
+            char === '/' ||
+            char === '.' ||
+            char === '{' ||
+            char === '}'
+        ) {
+            copy[i] = '-';
+        } else {
+            copy[i] = char;
+        }
+    }
+
+    return copy.join('');
```

Comment:
I believe the prevailing wisdom is that string concatenation is faster than this.